docs/specification/draft/basic/authorization.mdx

```diff
@@ -286,6 +330,50 @@ Servers **MUST** return appropriate HTTP status codes for authorization errors:
 | 403         | Forbidden    | Invalid scopes or insufficient permissions |
 | 400         | Bad Request  | Malformed authorization request            |
 
+#### Scope Error Handling
+
+This section covers handling insufficient scope errors during runtime operations when a client already has a token but needs additional permissions. This follows the error handling patterns defined in [OAuth 2.1 Section 5](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-13#section-5) and leverages the metadata fields from [RFC 9728 (OAuth 2.0 Protected Resource Metadata)](https://datatracker.ietf.org/doc/html/rfc9728).
+
+##### Runtime Insufficient Scope Errors
+
+When a client makes a request with an access token with insufficient scope during runtime operations, the server **SHOULD** respond with:
+
+- `HTTP 403 Forbidden` status code (per [RFC 6750 Section 3.1](https://datatracker.ietf.org/doc/html/rfc6750#section-3.1))
+- `WWW-Authenticate` header with the `Bearer` scheme and additional parameters:
+  - `error="insufficient_scope"` - indicating the specific type of authorization failure
+  - `scope="required_scope1 required_scope2"` - specifying the minimum scopes needed for the operation
+  - `error_description` (optional) - human-readable description of the error
+
+Example insufficient scope response:
+
+```http
+HTTP/1.1 403 Forbidden
+WWW-Authenticate: Bearer error="insufficient_scope",
+                         scope="files:read files:write",
+                         error_description="Additional file write permission required"
+```
+
+##### Client Scope Upgrade Flow
+
+When a client receives any scope-related error (initial authorization or runtime insufficient scope),
+it **SHOULD** follow this upgrade flow:
+
+1. **Parse error information** from the authorization server response or `WWW-Authenticate` header
+2. **Determine required scopes** using the following algorithm:
+   - If the `scope` parameter is present in the `WWW-Authenticate` header, use those scopes
+   - If `scope` is not available, use all scopes listed in `scopes_supported` from the Protected Resource Metadata document
+3. **Validate scope upgrade** to prevent infinite loops:
```

Comment:
This rules out a possibility for timed access (or some other condition), where the scope is no longer available for a given access token, but reauthorizing would re-establish the scope.  It feels a bit like "second guessing" the server a bit, and the "limited retry" logic below would be enough to detect a truly broken setup.